theano/gpuarray/dnn.py

```diff
@@ -2738,6 +2738,336 @@ def dnn_batch_normalization_test(inputs, gamma, beta, mean, var,
     return result
 
 
+class GpuDnnTransformerDesc(COp):
+    """
+    Descriptor Op for cuDNN Spatial Transformer.
+    """
+    __props__ = ('precision',)
+    params_type = ParamsType(precision=cudnn.cudnnDataType_t)
+
+    def c_headers(self):
+        return ['cudnn.h', 'cudnn_helper.h']
+
+    def c_header_dirs(self):
+        header_dirs = [os.path.dirname(__file__)]
+        if config.dnn.include_path:
+            header_dirs += [config.dnn.include_path]
+        return header_dirs
+
+    def c_libraries(self):
+        return ['cudnn']
+
+    def c_lib_dirs(self):
+        lib_dirs = []
+        if config.dnn.library_path:
+            lib_dirs += [config.dnn.library_path]
+        return lib_dirs
+
+    def do_constant_folding(self, node):
+        return False
+
+    def __init__(self, precision=theano.config.floatX):
+        COp.__init__(self, ["c_code/dnn_sptf_desc.c"], "APPLY_SPECIFIC(dnn_sptf_desc)")
+        assert cudnn.cudnnDataType_t.has_alias(precision)
+        self.precision = precision
```

Comment:
Ok.
@joaovictortr, so, I suggest this new scheme, in which we have no `GpuDnnTransformerDesc` anymore:
```python
# Recall: out_dims can be inferred as long as we have img and grid:
out_dims
	# out_dims = N,C,H,W:
	# N: img.shape[0], grid.shape[0]
	# C: img.shape[1]
	# H: grid.shape[1]
	# W: grid.shape[2]

GpuDnnTransformerGrid(theta, out_dims) -> grid
	# Pass out_dims instead of desc here and create desc in C code.

GpuDnnTransformerSampler(img, grid) -> transformed
	# out_dims can be inferred in C code (as we have img and grid).

GpuDnnTransformerGradI(img, grid, dy) -> dimg, dgrid
	# out_dims can be inferred in C code (as we have img and grid).

GpuDnnTransformerGradT(dgrid) -> dtheta
	# From dgrid we have N, H and W, but we miss C (number of channels).
	# Does this gradient really need to know C to compute dtheta ?
	# Could you try to use C == 1 to create the cuDNN desc, and check if it works ?
```